pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -359,6 +367,112 @@ func (d *defaultDriver) newElasticsearchClient(
 	)
 }
 
+// maybeSetServiceAccountsOrchestrationHint attempts to update an orchestration hint to let the association controllers
+// know whether all the nodes in the cluster are ready to authenticate service accounts.
+func (d *defaultDriver) maybeSetServiceAccountsOrchestrationHint(
+	ctx context.Context,
+	esReachable bool,
+	securityClient esclient.SecurityClient,
+	resourcesState *reconcile.ResourcesState,
+) error {
+	if d.ReconcileState.OrchestrationHints().ServiceAccounts.IsTrue() {
+		// Orchestration hint is already set to true, there is no point going back to false.
+		return nil
+	}
+
+	// Case 1: New cluster, we can immediately set the orchestration hint.
+	if !bootstrap.AnnotatedForBootstrap(d.ES) {
+		allNodesRunningServiceAccounts, err := esv1.AreServiceAccountsSupported(d.ES.Spec.Version)
+		if err != nil {
+			return err
+		}
+		d.ReconcileState.UpdateOrchestrationHints(
+			d.ReconcileState.OrchestrationHints().Merge(hints.OrchestrationsHints{ServiceAccounts: optional.NewBool(allNodesRunningServiceAccounts)}),
+		)
+		return nil
+	}
+
+	// Case 2: This is an existing cluster, but actual cluster version does not support service accounts.
+	if d.ES.Status.Version == "" {
+		return nil
+	}
+	supportServiceAccounts, err := esv1.AreServiceAccountsSupported(d.ES.Status.Version)
+	if err != nil {
+		return err
+	}
+	if !supportServiceAccounts {
+		d.ReconcileState.UpdateOrchestrationHints(
+			d.ReconcileState.OrchestrationHints().Merge(hints.OrchestrationsHints{ServiceAccounts: optional.NewBool(false)}),
+		)
+		return nil
+	}
+
```

Comment:
Should this be: 

```diff
        if !supportServiceAccounts {
                d.ReconcileState.UpdateOrchestrationHints(
-                       d.ReconcileState.OrchestrationHints().Merge(hints.OrchestrationsHints{ServiceAccounts: optional.NewBool(false)}),
+                       d.ReconcileState.OrchestrationHints().Merge(hints.OrchestrationsHints{ServiceAccounts: nil}),
                )
                return nil
        }
 
+       d.ReconcileState.UpdateOrchestrationHints(d.ReconcileState.OrchestrationHints().Merge(hints.OrchestrationsHints{
+               ServiceAccounts: optional.NewBool(false),
+       }))
+
```

So basically saying: we interpret the orchestration hint as follows: 
* `nil` i.e. annotation absent. ES is running an older version that does not support service accounts or the operator version is older 
* `false` there are service accounts configured but they are not observable yet on all nodes via the `_security` API
* `true` there are service accounts and the token are reported from all nodes